jax/experimental/compilation_cache/file_system_cache.py

```diff
@@ -37,6 +39,34 @@ def put(self, key: str, value: bytes):
     """Adds new cache entry, possibly evicting older entries."""
     if not key:
       raise ValueError("key cannot be empty")
-      #TODO(colemanliyah):implement eviction
-    with open(os.path.join(self._path, key), "wb") as file:
-      file.write(value)
+    if self._evict_entries_if_necessary(key, value):
+      path_to_new_file = os.path.join(self._path, key)
+      with open(path_to_new_file, "wb") as file:
+        file.write(value)
+    else:
+      warnings.warn(f"Cache value of size {len(value)} is larger than"
+                    f" the max cache size of {self._max_cache_size_bytes}")
+
+  def _evict_entries_if_necessary(self, key: str, value: bytes) -> bool:
+    """Returns True if there's enough space to add 'value', False otherwise."""
+    new_file_size = len(value)
+
+    if new_file_size >= self._max_cache_size_bytes:
+      return False
+
+    while new_file_size + self._get_cache_directory_size() > self._max_cache_size_bytes:
```

Comment:
this walks the entire directory on every while loop condition..it's a bit expensive

Reply:
I agree this is expensive, but I think we should improve it after we have things working end-to-end and have microbenchmarks (so we can see how much it helps!). Maybe leave a TODO here though.